src/models/string-or-translations.ts

```diff
@@ -2,18 +2,39 @@ import Joi from 'joi';
 import { schema as translationsSchema, Translations } from './translations';
 import { Locale } from './locale';
 
-const schema = Joi.alternatives().try(
-  Joi.string().example('untranslated string').meta({ className: 'untranslatedString' }),
-  translationsSchema,
-)
-  .tag('stringOrTranslations')
-  .meta({ className: 'stringOrTranslations' });
+const schema = (limit?: number): Joi.AnySchema => {
+  if (limit !== undefined) {
+    return Joi.alternatives().try(
+      Joi.string().example('untranslated string').meta({ className: 'untranslatedString' }).max(limit),
+      translationsSchema(limit),
+    )
+      .tag('stringOrTranslations')
+      .meta({ className: 'stringOrTranslations' });
+  }
+
+  return Joi.alternatives().try(
+    Joi.string().example('untranslated string').meta({ className: 'untranslatedString' }),
+    translationsSchema(),
+  )
+    .tag('stringOrTranslations')
+    .meta({ className: 'stringOrTranslations' });
+};
 
-const stringBeforeV7ElseStringOrTranslationSchema = (apiVersion: number): Joi.AnySchema => {
-  if (apiVersion <= 6) {
-    return Joi.string();
+const versionedStringOrStringOrTranslationSchema = (apiVersion: number): Joi.AnySchema => {
```

Comment:
As discussed I went back to the simpler version of the function since there is no need to support localised names longer than 255 chars fb01c6e35b0bc43b29669518d2a0aa4370fe0317